packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,650 @@
+    }
+
+    if (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments !== undefined
+      && !this.isArrayOfStrings(props.jobDriver.sparkSubmitJobDriver.entryPointArguments.value)
+      && !sfn.JsonPath.isEncodedJsonPath(props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value)
+      && (props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value.length > 10280
+          || props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value.length < 1)) {
+      throw new Error(`Entry point arguments must be an string array between 1 and 10280 in length. Received ${props.jobDriver.sparkSubmitJobDriver?.entryPointArguments.value.length}.`);
+    }
+
+    if (props.jobDriver.sparkSubmitJobDriver?.sparkSubmitParameters !== undefined
+      && (props.jobDriver.sparkSubmitJobDriver.sparkSubmitParameters.length > 102400
+        || props.jobDriver.sparkSubmitJobDriver.sparkSubmitParameters.length < 1)) {
+      throw new Error(`Spark submit parameters must be between 1 and 102400 characters in length. Received ${props.jobDriver.sparkSubmitJobDriver.sparkSubmitParameters.length}.`);
+    }
+
+    if (props.executionRole === undefined
+      && sfn.JsonPath.isEncodedJsonPath(props.virtualClusterId.value)) {
+      throw new Error('Execution role cannot be undefined when the virtual cluster ID is not a concrete value. Provide an execution role with the correct trust policy');
+    }
+
+    this.logGroup = this.props.monitoring?.logGroup ?? this.props.monitoring?.logging ? new logs.LogGroup(this, 'Monitoring Log Group') : undefined;
+    this.logBucket = this.props.monitoring?.logBucket ?? this.props.monitoring?.logging ? new s3.Bucket(this, 'Monitoring Bucket') : undefined;
+    this.role = this.props.executionRole ?? this.createJobExecutionRole();
+    this.taskPolicies = this.createPolicyStatements();
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('emr-containers', 'startJobRun', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        VirtualClusterId: this.props.virtualClusterId.value,
+        Name: this.props.jobName,
+        ExecutionRoleArn: this.role.roleArn,
+        ReleaseLabel: this.props.releaseLabel.label,
+        JobDriver: {
+          SparkSubmitJobDriver: {
+            EntryPoint: this.props.jobDriver.sparkSubmitJobDriver?.entryPoint.value,
+            EntryPointArguments: this.props.jobDriver.sparkSubmitJobDriver?.entryPointArguments?.value,
+            SparkSubmitParameters: this.props.jobDriver.sparkSubmitJobDriver?.sparkSubmitParameters,
+          },
+        },
+        ConfigurationOverrides: {
+          ApplicationConfiguration: cdk.listMapper(this.applicationConfigPropertyToJson)(this.props.applicationConfig),
+          MonitoringConfiguration: {
+            CloudWatchMonitoringConfiguration: {
```

Comment:
A similar thing to the `s3 monitoring configuration` occurs here too,  the `logGroupName` property is required if cloudWatchMonitoringConfiguration is defined.